python/add_corona.py

```diff
@@ -0,0 +1,173 @@
+            continue
+        # particle line
+        else:
+            particle = np.array(line[:12])
+            particle = np.append(int(n_event), particle)
+            particles = np.append(particles,[particle], axis=0)
+    return(particles)
+
+def gather_corona_particles(corona_lists):
+    '''
+        Extract particle lines from the initial conditions oscar output of SMASH
+        and any extra file (e.g. from vHLLE oscar output) and join particles
+        into one array.
+    '''
+    is_any_corona_file_present = False
+    corona_particles = np.empty(shape=[0, 13])
+    for file in corona_lists:
+        if os.path.isfile(file):
+            is_any_corona_file_present = True
+            more_corona_particles = extract_particles(file)
+            corona_particles = np.append(corona_particles, more_corona_particles, axis=0)
+    if len(corona_particles) == 0:
+        print("No corona particles!", file=sys.stderr)
+        sys.exit(4)
+    if not is_any_corona_file_present:
+        print("Corona files not found!", file=sys.stderr)
+        sys.exit(2)
+    return corona_particles
+
+def copy_header(input_file, output_file):
+    header = ""
+    for line in open(input_file, "r"):
+        if line.startswith('# event'):
+            break
+        else:
+            header += line
+    with open(output_file, 'w') as f:
+        f.write(header)
+
+def read_sampled_and_write_full_particle_list(args, corona_particles, n_events_ic):
+    '''
+        Corona particles are distributed among sampled events
+    '''
+    output_file = args.output_file
+    copy_header(args.sampled_particle_list,output_file)
+
+    sampled_particles = np.empty(shape=[0, 12])
+    event_s = 0
+    event_c = 0
+
+    for line in open(args.sampled_particle_list, "r"):
+        line = line.split()
+        # event end line
+        if "end" in line:
+            corona_filter = corona_particles[corona_particles[:, 0] == str(event_c)]
+            particle_number = len(corona_filter)+len(sampled_particles)
+            with open(output_file, 'a') as f:
+                f.write("# event {} out {}\n".format(event_s,particle_number))
+            # write sampled
+            with open(output_file, 'a') as f:
+                np.savetxt(f, sampled_particles, delimiter=' ', fmt='%s')
+            # write corona
+            with open(output_file, 'a') as f:
+                np.savetxt(f, corona_filter[:,1:], delimiter=' ', fmt='%s')
+            with open(output_file, 'a') as f:
+                f.write("# event {} end \n".format(event_s))
+
+            if event_s % 100 == 0:
+                print("read and write event "+str(event_s))
+            # reset relevant variables
+            if event_c < (n_events_ic-1):
+                event_c += 1
+            else:
+                event_c = 0
+            event_s += 1
+            sampled_particles = np.empty(shape=[0, 12])
+        # ignore event headers
+        elif "#" in line[0]:
```

Comment:
I'd use here the same safety check to skip empty lines, to avoid an `IndexError`.
```suggestion
        elif len(line) == 0 or "#" in line[0]:
```